### PR TITLE
Provisioner kind -> ClusterProvisioner

### PR DESCRIPTION
**Pull request by scothis**
There is no Provisioner CRD in the current implementation. The CRD is
named ClusterProvisioner. "Provisioner" is still a concept and other
references in the spec are able to handle either a Provisioner or
ClusterProvisioner in the future.

Supersedes #6

cc @matzew

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ This document details our _Spec_ and _Status_ customizations.
 - [Source](#kind-source)
 - [Channel](#kind-channel)
 - [Subscription](#kind-subscription)
-- [Provider](#kind-provisioner)
+- [ClusterProvisioner](#kind-clusterprovisioner)
```
Comment (matzew) [inline docs/spec/spec.md:14]: :+1:

**Comment by scothis** [comment]
@matzew @sjwoodman I updated this PR with the latest from n3wscott/push_raw_spec. Please take a look

File: docs/spec/overview.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ objects. The interfaces are ([Sinkable](interfaces.md#sinkable),
 ## Subscription
 
 **Subscriptions** describe a flow of events from one _Channel_) to the next
-Channel* through transformations (such as a Knative Service which processes
+Channel\* through transformations (such as a Knative Service which processes
```
Comment (matzew) [inline docs/spec/overview.md]: `\` ?

**Comment by n3wscott** [comment]
Did it with search and replace...

**Comment by scothis** [comment]
@n3wscott references to `Provider` still need to be updated

**Comment by n3wscott** [comment]
my intent is that `provisioner` == `ClusterChannelProvisioner`

**Pull request closed without merging**